### PR TITLE
Add property transform for contract tests

### PR DESCRIPTION
**Pull request by zjinmei**
*Description of changes:*
When resource handlers input and output are not equal in contract tests, we can utilize property transform for input model and compare with output model again, if succeed, CT pass, if failed, CT failed

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/rpdk/core/contract/suite/handler_commons.py [hunk 1]
```diff
@@ -164,7 +164,11 @@ def test_input_equals_output(resource_client, input_model, output_model):
     pruned_output_model = prune_properties_if_not_exist_in_path(
         pruned_output_model, pruned_input_model, resource_client.create_only_paths
     )
+
+    transformed_input_model = resource_client.transform_model(
+        copy.deepcopy(pruned_input_model)
+    )
+
```
Comment (ammokhov) [inline src/rpdk/core/contract/suite/handler_commons.py:171]: do you think we need additional copy of it? can we just use transform during the comparison and not as additional input?

Comment (zjinmei) [reply src/rpdk/core/contract/suite/handler_commons.py]: Hard to use transform during the comparison as pyjq needs inputs with whole model. But updated the function with less inputs and do copy in resource_client.py

File: src/rpdk/core/contract/resource_client.py [hunk 2]
```diff
@@ -225,13 +226,35 @@ def _update_schema(self, schema):
...
 
         additional_identifiers = self._schema.get("additionalIdentifiers", [])
         self._additional_identifiers_paths = [
             {fragment_decode(prop, prefix="") for prop in identifier}
             for identifier in additional_identifiers
         ]
 
+    def transform_model(self, input_model):
+        if not self.property_transform or sys.platform.startswith("win"):
+            return None
```
[the remark this answers is not in the document]
Comment (zjinmei) [reply src/rpdk/core/contract/resource_client.py:241]: Updated with Environment exception here

File: README.md [hunk 3]
```diff
@@ -59,7 +59,31 @@ cfn test --enforce-timeout 60 -- -k contract_delete_update # combine args
 cfn test --log-group-name cw_log_group --log-role-arn log_delivery_role_arn # Handler logs generated by contract tests will be delivered to the specified cw_log_group using the credentials from log_delivery_role_arn
 ```
 
-Note: To use your type configuration in contract tests, you will need to save your type configuration json file in `~/.cfn-cli/typeConfiguration.json`.
+Note:
+* To use your type configuration in contract tests, you will need to save your type configuration json file in `~/.cfn-cli/typeConfiguration.json`.
+
+* When the resource type with propertyTransform in schema, need to install PYJQ firstly(propertyTransform for contract tests is not available for Windows system)
+
```
[the remark this answers is not in the document]
Comment (zjinmei) [reply README.md]: Updated, thanks

File: src/rpdk/core/contract/resource_client.py [hunk 4]
```diff
@@ -225,13 +227,42 @@ def _update_schema(self, schema):
...
+
+        for key in self.property_transform_keys:
+            path = "/" + "/".join(key)
+            expression = self.property_transform[path]
+            transformed_value = pyjq.first(expression, input_model)
+            input_model = self.update_property(input_model, transformed_value, key[1:])
+
+        return input_model
+
+    def update_property(self, model, value, path):
```
Comment (ammokhov) [inline src/rpdk/core/contract/resource_client.py:259]:
```python
def update_property(self, model: dict, value: ..., path: tuple) -> dict: 
```

Comment (zjinmei) [reply src/rpdk/core/contract/resource_client.py]: Updated, thank you

**Review by ammokhov — APPROVED** [review]

**Review by omkhegde — APPROVED** [review]

**Review by mmaeng — APPROVED** [review]

**Pull request merged**